### ISSUE TITLE
Boost doesn't compile on 32 bit processor

### ISSUE THREAD

**Issue by Astalaseven** [fw4spl-org/fw4spl-deps#2]
Hi,

the `Makefile` tries to install boost but it's configured to compile it for a 64 bit system.
As everyone doesn't have a 64 bit system, could it be possible to check it? The error is not easy to find otherwise.

Quick patch:

``` diff
diff --git a/boost/CMakeLists.txt b/boost/CMakeLists.txt
index a321128..d5faf78 100644
--- a/boost/CMakeLists.txt
+++ b/boost/CMakeLists.txt
@@ -28,7 +28,7 @@ set(BOOST_ARGS
     --without-test
     variant=${BOOST_BUILD_TYPE}
     link=shared
-    address-model=64
+    address-model=32
     threading=multi
 )
```

**Comment by FredChamp** [comment]
Hi Astalaseven,

Are you compiling every dependencies in 32 bits ?
Because nobody have 32 bits system at the office then it will be complicated to find an elegant way to fix this issue but we will try ;)

**Comment by Astalaseven** [comment]
Yes I am, it's currently the only library where the problem occurs. Perhaps removing the `address-model` will take the one needed without specifying it?

**Comment by schweitzer** [comment]
Hi,
A simple solution could be the test of the CMake flag "CMAKE_SIZEOF_VOID_P"  which will be 8 on 64 bits system and 4 on 32 bits ones.

**Comment by greenjava** [comment]
Hi
Good idea @schweitzer :)
To be portable, we can write something like that :

``` patch
diff -r ab715a94d8d0 boost/CMakeLists.txt
--- a/boost/CMakeLists.txt  Thu Jul 16 09:28:43 2015 +0200
+++ b/boost/CMakeLists.txt  Fri Jul 17 08:27:06 2015 +0200
@@ -16,6 +16,7 @@
 set(PYTHON_INCLUDE ${PYTHON_DIR}/include/python2.7)
 set(BOOST_PATCH_DIR ${CMAKE_CURRENT_SOURCE_DIR}/patch)

+math(EXPR BITS "8*${CMAKE_SIZEOF_VOID_P}")
 set(BOOST_ARGS
     --exec-prefix=${CMAKE_INSTALL_PREFIX}/bin
     -sZLIB_INCLUDE=${CMAKE_INSTALL_PREFIX}/include
@@ -28,7 +29,7 @@
     --without-test
     variant=${BOOST_BUILD_TYPE}
     link=shared
-    address-model=64
+    address-model=${BITS}
     threading=multi
 )
```

**Comment by greenjava** [comment]
Issue resolved in 9c45ecefa11ae11b5d06a78a683f432d1f26043b
Thank you for reporting this issue
